gettsim/interface.py

```diff
@@ -65,13 +69,87 @@ def compute_taxes_and_transfers(
         DataFrame containing computed variables.
 
     """
+
+    # Set defaults for some parameters
     targets = DEFAULT_TARGETS if targets is None else targets
     targets = parse_to_list_of_strings(targets, "targets")
     columns_overriding_functions = parse_to_list_of_strings(
         columns_overriding_functions, "columns_overriding_functions"
     )
     params = {} if params is None else params
 
+    # Create the dag
+    dag = set_up_dag(
+        data,
+        params,
+        functions,
+        targets,
+        columns_overriding_functions,
+        check_minimal_specification,
+        rounding,
+    )
+
+    # We delay the data preparation as long as possible such that other checks can fail
+    # before this
+    data = data.copy(deep=True)
+    data = _process_data(data)
+    data = _reduce_data(data)
+    ids = _dict_subset(data, set(data) & {"hh_id", "tu_id"})
+
+    results = execute_dag(dag, data, targets, debug)
+
+    results = _expand_data(results, ids)
+    results = pd.DataFrame(results)
+
+    if not debug:
+        results = results[targets]
+
+    results = _reorder_columns(results)
+
+    return results
+
+
+def set_up_dag(
```

Comment:
Only realised this just now: It is pretty confusing to have one function `set_up_dag`, which calls `create_dag`. Can we do better? No time for brainstorming right now, sorry.